### DISCUSSION TITLE
Generative model and NLP

### DISCUSSION THREAD

**Discussion by agyeong** [Boostcamp-AI-Tech-1-15/ddobokki_log#16] [2. DL Basic]
강의의 예시 자료에 텍스트는 왜 없을까라는 의문에서 시작했습니다.
텍스트를 어떻게 적용할 수 있을까요?

문장을 넣으면 generator가 위조문장(?)을 만들고 discriminator가 Janson-Shannon Divergence를 최소화시킨다면 어떤 결과가 나올지 궁금하네요. GAN은 데이터 분포를 만드는 것이라고 이해했는데 특정 주제를 갖는 문장들을 학습시켰을 때 생기는 분포들을 비교하면 어떨까,,,라는 생각이 들었습니다. 

질문을 정리하고 나니까 뜬구름 잡는 얘기처럼 들리네요... :(

**Comment by mybirth0407** [comment]
GAN은 NLP에서 naive 하게 사용될 수 없는데요, https://kh-kim.gitbook.io/natural-language-processing-with-pytorch/00-cover-11/01-intro 를 참고하시면 좋을 것 같습니다.
결론적으로는 확률 분포를 근사해야 할건데 그 방법으로 대표적인 게 reinforcement learning이 있습니다.
-> importance sampling과 같은 개념을 이용하여 어떤 구하고자 하는 q라는 어려운 확률 분포를 알고 있는 쉬운 p로 근사하는 방식인데, 이론상으로는 깔끔하나 굉장히 어렵습니다..!
